### PR TITLE
Get an Array of Cookies

### PR DESCRIPTION
**Pull request by carsteneu**
We added 2 methods to get an Array of used Cookies of a given page and an Array of the names of these Cookies, for this purpose we added a method to get the whole raw data from the Cookie Class.

`$cookies->getListNamesAsArray();`
`$cookies->getListAllDataAsArray();`

This is quite handy if you want to have a Look at the Cookies a page ist using.

Also added .idea for the gitignore - setting files from jetbrains IDEs.

### CODE REVIEW

**Comment by enricodias** [comment]
Thanks for the PR.

Since the `Cookie::class` already implements `ArrayAccess`, whats the point of `getListAllDataAsArray()`?

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,4 @@ composer.lock
 phpstan.neon
 phpunit.xml
 vendor
+.idea
```
Comment (GrahamCampbell) [inline .gitignore]: Please don't add this. Put this in your global ignore file.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

**Comment by GrahamCampbell** [comment]
Can you enable editing by maintainers, so I can clean this up, please?

File: .gitignore (hunk 1 above)
Comment (carsteneu) [reply .gitignore]: cant see the checkbox for "editing by maintainers"

**Comment by carsteneu** [comment]
> Thanks for the PR.
> 
> Since the `Cookie::class` already implements `ArrayAccess`, whats the point of `getListAllDataAsArray()`?

I thought it might be handy... i use it in my project in this way.

**Comment by GrahamCampbell** [comment]
Going to close this, since this is already possibly to implement in apps without modifying our code.

**Pull request closed without merging**